src/main/java/entity/User.java

```diff
@@ -61,6 +64,15 @@ public User() {}
     public void setRecipes(List<Recipe> recipes) {this.recipes = recipes;}
     public void setShoppingLists(List<ShoppingList> shoppingLists) {this.shoppingLists = shoppingLists;}
     public void setPreference(Map<String, Object> preference) {this.preference = preference;}
+    public void addRecentlyViewedRecipe(Recipe recipe) {
+        if (this.recentlyViewedRecipes.size() >= 5) {
+            this.recentlyViewedRecipes.remove(0); // one sec
+            this.recentlyViewedRecipes.add(recipe);
+        }
+        else {
+            this.recentlyViewedRecipes.add(recipe);
```

Comment:
if (this.recentlyViewedRecipes.size() >= 5) {
            this.recentlyViewedRecipes.remove(0);
}
this.recentlyViewedRecipes.add(recipe);

Reply:
changed to this